src/homework9/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,170 @@
+
+        public void setPrev(Node<T> prev) {
+            this.prev = prev;
+        }
+
+        public Node<T> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<T> next) {
+            this.next = next;
+        }
+
+        public T getElement() {
+            return element;
+        }
+
+        public void setElement(T element) {
+            this.element = element;
+        }
+    }
+
+    public LinkedList() {
+        size = 0;
+    }
+
+    @Override
+    public void add(T value) {
+        Node<T> node = new Node<>(last, value, null);
+        if (size == 0) {
+            first = node;
+        } else {
+            last.setNext(node);
+        }
+        last = node;
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index > -1 && index <= size) {
+            if (index == size) {
+                add(value);
+            } else {
+                Node<T> necessaryNode = getNodeByIndex(index);
+                Node<T> node = new Node<>(necessaryNode.getPrev(), value, necessaryNode);
+                necessaryNode.getPrev().setNext(node);
+                necessaryNode.setPrev(node);
+                size++;
+            }
+        } else {
+            throw new NoSuchElementException();
+        }
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        if (!list.isEmpty()) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        checkIndex(index);
+        return getNodeByIndex(index).getElement();
+    }
+
+    @Override
+    public void set(T value, int index) {
+        checkIndex(index);
+        getNodeByIndex(index).setElement(value);
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public T remove(int index) {
+        Node<T> currentNode = getNodeByIndex(index);
+        Node<T> prevNode = currentNode.getPrev();
+        Node<T> nextNode = currentNode.getNext();
+
+        if (index < size - 1 && index > 0) {
+            prevNode.setNext(nextNode);
+            nextNode.setPrev(prevNode);
+            currentNode.setPrev(null);
+            currentNode.setNext(null);
+            return currentNode.getElement();
+        } else if (index == 0) {
+            nextNode.setPrev(null);
+            first = nextNode;
+            return currentNode.getElement();
+        } else if (index == size - 1) {
+            prevNode.setNext(null);
+            last = prevNode;
+            return currentNode.getElement();
+        }
+        size--;
+        throw new NoSuchElementException();
+    }
+
+    @Override
+    public T remove(T t) {
+        if (t != null) {
+            for (int i = 0; i < size; i++) {
+                if (t.equals(getNodeByIndex(i).getElement())) {
```

Comment:
method get(int index) has a for loop inside so here we have loop inside of loop. This method complexity is O(n^2) we can implement this method with one loop.

Reply:
Ok. I did.